kdTree.cpp

```diff
@@ -0,0 +1,344 @@
+#include "precomp.h"
```

Comment:
Function names do not match the established style.
i.e.: 
kdTree::insertnodes  should be insert_nodes
kdTree::searchClosest should be search_closest